pylint/checkers/variables.py

```diff
@@ -979,267 +1001,296 @@ def visit_assignname(self, node: nodes.AssignName) -> None:
     def visit_delname(self, node: nodes.DelName) -> None:
         self.visit_name(node)
 
-    # pylint: disable=too-many-branches
+    @utils.check_messages(
+        "undefined-variable",
+        "used-before-assignment",
+        "cell-var-from-loop",
+        "unused-import",
+    )
     def visit_name(self, node: nodes.Name) -> None:
-        """Check that a name is defined in the current scope"""
         stmt = node.statement()
         if stmt.fromlineno is None:
             # name node from an astroid built from live code, skip
             assert not stmt.root().file.endswith(".py")
             return
 
+        self._undefined_and_used_before_checker(node, stmt)
+        self._loopvar_name(node)
```

Comment:
The first thing `self._loopvar_name` does is check if the message type is enabled. Could you move the check here? That way we won't even have to call the function if it's disabled. (The check itself could also be moved to `def open()`.)

https://github.com/PyCQA/pylint/blob/6caab13816e8098063b6e077ab0044b94aa87557/pylint/checkers/variables.py#L1710-L1711

Reply:
I think we normally don't put these kinds of checks in `open`? For maintainability I think it makes sense to do the if-check at the same place at which the actual check is started. (Your suggestion improves on this)